File: java/mission-to-mars-solid/src/main/java/mars/Rover.java
package mars;

import lombok.Getter;

public class Rover implements ModuleMartien2D {

	private Direction direction;
	@Getter
	protected Position position;
	@Getter
	private boolean pretPourRecuperation = false;
	
	public Rover(Direction direction, Position position) {
		this.direction = direction;
		this.position = position;
	}

	public void avancer() {
		position = direction.getMouvement2D().avancer(position);
		//position = new DirectionHandlerFactory().apply(direction).avancer(position);
	}

	public void reculer() {
		position = direction.getMouvement2D().reculer(position);
		//position = new DirectionHandlerFactory().apply(direction).reculer(position);
	}

	public void activerRecuperation() {
		pretPourRecuperation = true;
	}

}

File: python/mission-to-mars-solid/src/mars/Rover.py
from src.mars.Direction import Direction
from src.mars.ModuleMartien2D import ModuleMartien2D
from src.mars.Position import Position


class Rover(ModuleMartien2D):
    def __init__(self, direction: Direction, position: Position):
        super().__init__(direction, position)

    def activer_recuperation(self) -> None:
        self._pret_pour_recuperation = True
